Woof/Woof/NetworkLayer/Endpoint/Encodable+asDictionary.swift

```diff
@@ -0,0 +1,18 @@
+import Foundation
+
+extension Encodable {
+    /// Converts the object to a dictionary representation.
+    ///
+    /// - Throws: An error of type `Error` if encoding or serialization fails.
+    ///
+    /// - Returns: A dictionary representation of the encoded object.
+    func asDictionary() throws -> [String: Any] {
+        let encoder = JSONEncoder()
+        encoder.keyEncodingStrategy = .convertToSnakeCase
+
+        let data = try encoder.encode(self)
+        let dictionary = try JSONSerialization.jsonObject(with: data, options: .allowFragments) as? [String: Any]
+
+        return dictionary ?? [:]
+    }
```

Comment:
I think we should return an optional dictionary to a caller. The empty dictionary is more ambiguous.
```suggestion
    func asDictionary() throws -> [String: Any]? {
        let encoder = JSONEncoder()
        encoder.keyEncodingStrategy = .convertToSnakeCase

        let data = try encoder.encode(self)
        let dictionary = try JSONSerialization.jsonObject(with: data, options: .allowFragments) as? [String: Any]

        return dictionary
    }
```

Reply:
Ok, we can return the optional dictionary. I think we can remove `throw` for this method and return `nil` if the transformation fails.